From aebb91551b3781f48d2eae3bc94178d3b604d573 Mon Sep 17 00:00:00 2001
From: Hanna Hawa <hannah@marvell.com>
Date: Wed, 26 Aug 2015 07:38:47 +0300
Subject: [PATCH 0351/1240] fix: ap806: updated SPI frequency for AP-806 DB
 board

- Keep the SPI frequency for Palladuim AS-IS

Change-Id: I7d809cff1c721f867699ee176c08d82ce49e71e3
Signed-off-by: Hanna Hawa <hannah@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/23188
Tested-by: Star_Automation <star@marvell.com>
---
 arch/arm/dts/apn-806-pd.dts | 9 +++++++++
 arch/arm/dts/apn-806.dtsi   | 2 +-
 2 files changed, 10 insertions(+), 1 deletion(-)

diff --git a/arch/arm/dts/apn-806-pd.dts b/arch/arm/dts/apn-806-pd.dts
index 46d2aa1..7c3422e 100644
--- a/arch/arm/dts/apn-806-pd.dts
+++ b/arch/arm/dts/apn-806-pd.dts
@@ -24,6 +24,15 @@
 	model = "Marvell APN-806 Palladuim";
 	compatible = "marvell,apn-806-pd", "marvell,apn-806";
 
+	soc {
+
+		internal-regs {
+
+			spi0: spi@510600 {
+				spi-max-frequency = <51200>;
+			};
+		};
+	};
 };
 
 &pinctl {
diff --git a/arch/arm/dts/apn-806.dtsi b/arch/arm/dts/apn-806.dtsi
index 4abf701..59db3b8 100644
--- a/arch/arm/dts/apn-806.dtsi
+++ b/arch/arm/dts/apn-806.dtsi
@@ -56,7 +56,7 @@
 			spi0: spi@510600 {
 				compatible = "marvell,orion-spi";
 				reg = <0x510600 0x50>;
-				spi-max-frequency = <51200>;
+				spi-max-frequency = <0x100000>;
 				clock = <&tclk>;
 				status = "okay";
 			};
-- 
1.9.1

